nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractUtils.java

```diff
@@ -123,15 +126,32 @@ static String[] getAntecedent(
   static @Nullable String getContractString(Symbol.MethodSymbol methodSymbol, Config config) {
     for (AnnotationMirror annotation : methodSymbol.getAnnotationMirrors()) {
       String name = AnnotationUtils.annotationName(annotation);
-      if (config.isContractAnnotation(name)) {
+      if (hasSimpleNameContract(name) || config.isContractAnnotation(name)) {
         return NullabilityUtil.getAnnotationValue(methodSymbol, name);
       }
     }
     return null;
   }
 
+  /**
+   * Checks if the input string, which is a qualified name, has simple name "Contract".
+   *
+   * @param input the qualified name to check
+   * @return true if the simple name is "Contract", false otherwise
+   */
+  private static boolean hasSimpleNameContract(String input) {
+    int lastDot = input.lastIndexOf('.');
```

Comment:
Do we ever want to support an inner class as a contract? E.g. `Foo$[Bar$]Contract`? Probably not for now, but just want to note this requires `Contract` to be a top-level class :)

Reply:
Actually I think inner classes happen to work already; see [here](https://github.com/uber/NullAway/blob/02221408699c6a9beef090c5ae411a9df7033a4f/nullaway/src/test/java/com/uber/nullaway/ContractsTests.java#L91).  I think the way that `javac` works we see source-level names, not bytecode-level, so `$` doesn't get used.  In any case, I don't think we want to go further out of our way to support this case :-)

Reply:
Ah, right. Missed that we are already testing that. The whole confusion in Java between `a.b.c.D.E` and `a.b.c.D$E` depending on what level your tool works at is a bit annoying :sweat_smile: